pkg/pdutil/placement_client.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pdutil
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/store/pdtypes"
+	"github.com/pingcap/tiflow/pkg/config"
+	"github.com/pingcap/tiflow/pkg/httputil"
+	"github.com/pingcap/tiflow/pkg/retry"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+)
+
+const (
+	updateTypeDel = iota
+	updateTypeAdd
+)
+
+const (
+	batchUpdatePrefix = "/pd/api/v1/config/rules/batch"
+	idQueryPrefix     = "/pd/api/v1/config/rule"
+
+	// `index` refers to the stacking sequence of rules in a group.
+	// `override` specifies whether to overwrite rules with smaller index (in a group).
+	addMetaJSON = `[{
+		"group_id": "pd",
+		"id": "cdcmeta",
+		"index": 20000,    
```

Comment:
Why choosing index value 2000?

Reply:
This is an arbitrary value large enough to ensure that it can override other rules. Similar to this:
https://docs.pingcap.com/tidb/stable/configure-placement-rules#scenario-5-migrate-a-table-to-the-tiflash-cluster